bots/task/distributed_queue.py

```diff
@@ -0,0 +1,75 @@
+# This file is part of Cockpit.
+#
+# Copyright (C) 2019 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Shared GitHub code. When run as a script, we print out info about
+# our GitHub interacition.
+
+import ssl
+
+import pika
+
+__all__ = (
+    'DistributedQueue',
+    'BASELINE_PRIORITY',
+    'MAX_PRIORITY',
+)
+
+BASELINE_PRIORITY = 5
+MAX_PRIORITY = 9
+
+arguments = {
+    'rhel': {
+        "x-max-priority": MAX_PRIORITY
+    },
+    'public': {
+        "x-max-priority": MAX_PRIORITY
+    },
+}
+
+class DistributedQueue(object):
+    # amqp_server should be formatted as host:port
+    # queues should be a list of strings with the names of queues
```

Comment:
OK for now, just for the future: The Pythonic way are [docstrings](https://www.python.org/dev/peps/pep-0257/), not comments. This isn't widely published API, so no biggie here.

This should document `kwargs` (like "passed to `queue_declare()`"), and explain what happens to the queue list.